### PR TITLE
Prepare the deployment to Bintray/Maven Central

### PR DESCRIPTION
**Pull request by malafeev**

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# which will call out to this script
+set -e
+
+# Use maven-help-plugin to get the current project.version
+VERSION=`mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -v '\['`
+
+echo "Publishing $VERSION"
+
+# Build and deploy to Bintray
+mvn -s .circleci.settings.xml -Dmaven.test.skip=true deploy -pl .,lightstep-tracer-jre,lightstep-tracer-jre-bundle,shadow
```
Comment (carlosalberto) [inline deploy.sh:14]: Wrong components ;) 

We have to use `mvn -s .circleci.settings.xml -Dmaven.test.skip=true deploy`

(I don't think we need to manually select what components we want to deploy, as we want to deploy all our 3 artifacts)

Comment (malafeev) [reply deploy.sh]: done

File: deploy.sh [hunk 2]
```diff
@@ -3,23 +3,18 @@
...
 
 echo "Publishing $VERSION"
 
 # Build and deploy to Bintray
-mvn deploy
+mvn -s .circleci.settings.xml -Dmaven.test.skip=true deploy
 
 # Sign the jar and other files in Bintray
-curl -H "X-GPG-PASSPHRASE:$BINTRAY_GPG_PASSPHRASE" -u $BINTRAY_USER:$BINTRAY_API_KEY -X POST https://api.bintray.com/gpg/lightstep/maven/otel-launcher-java/versions/$VERSION
+curl -H "X-GPG-PASSPHRASE:$BINTRAY_GPG_PASSPHRASE" -u $BINTRAY_USER:$BINTRAY_API_KEY -X POST "https://api.bintray.com/gpg/lightstep/maven/lightstep-tracer-jre/versions/$VERSION"
```
Comment (carlosalberto) [inline deploy.sh:17]: Also, this needs to refer to "otel-launcher-java" (as it was before this PR). Sorry, totally missed this in the previous review ;(

File: deploy.sh [hunk 3]
```diff
@@ -3,23 +3,18 @@
...
 
 # Sync the repository with Maven Central
-curl -H "Content-Type: application/json" -u $BINTRAY_USER:$BINTRAY_API_KEY -X POST -d '{"username":"'$MAVEN_CENTRAL_USER_TOKEN'","password":"'$MAVEN_CENTRAL_TOKEN_PASSWORD'","close":"1"}' https://api.bintray.com/maven_central_sync/lightstep/maven/otel-launcher-java/versions/$VERSION
+curl -H "Content-Type: application/json" -u $BINTRAY_USER:$BINTRAY_API_KEY -X POST -d '{"username":"'$MAVEN_CENTRAL_USER_TOKEN'","password":"'$MAVEN_CENTRAL_TOKEN_PASSWORD'","close":"1"}' "https://api.bintray.com/maven_central_sync/lightstep/maven/lightstep-tracer-jre/versions/$VERSION"
```
Comment (carlosalberto) [inline deploy.sh:20]: Same.

File: deploy.sh (hunk 2 above)
Comment (malafeev) [reply deploy.sh]: fixed

File: deploy.sh (hunk 3 above)
Comment (malafeev) [reply deploy.sh]: fixed

**Pull request merged**